Update Cody capabilities (#56269)
sourcegraph/sourcegraph: doc/cody/capabilities.md

# Cody capabilities

> <p class="subtitle">Learn more about Cody's features.</p>

Author: Yes, for sure. My goal with that change was not to overpromise the product while it's beta. I can make a new suggestion here.

> <p class="subtitle">Learn more about Cody's features.</p>

Author: ```suggestion
<p class="subtitle">Learn and understand more about Cody's features and core AI functionality.</p>
```

> <p class="subtitle">Learn more about Cody's features.</p>

Reviewer: This looks better. 👍

## Autocomplete

Cody suggests completions as you type using context from your code, such as your open files and file history. It’s powered by the latest instant LLM models for accuracy and performance.

Autocomplete supports any programming language because it uses LLMs trained on broad data. We've found that it works especially well with Python, Go, JavaScript, and TypeScript code.

![Example of Cody autocomplete. You see a code snippet starting with async function getWeather(city: string) { and Cody response with a multi-line suggestion using a public weather API to return the current weather ](https://storage.googleapis.com/sourcegraph-assets/website/Product%20Animations/GIFS/cody-completions-may2023-optim.gif)

> ### Configure autocomplete on Sourcegraph Enterprise

Author: It is treated as a license/SKU name. Today it is typically capitalized, and we could reconsider that, but we should make some changes for consistency then. Here's an example of how it's used today: https://handbook.sourcegraph.com/departments/product/process/gtm/licensing

> ### Configure autocomplete on Sourcegraph Enterprise

Reviewer: Having an "enterprise" aspect is pretty much a norm in the tech world. I'd suggest let's reconsider how we're naming it.

> ### Configure autocomplete on Sourcegraph Enterprise

Author: That's fair. For that kind of change, we should roll it out as a synchronized change (on docs, handbook) all together when we're ready. For now, can we rework this to avoid conflict?
```suggestion
### Configure autocomplete on an enterprise Sourcegraph instance
```

</div>

### Inline chat and code fixes

In VS Code, you can also open Cody chat inline using the `+` icon. This opens a chat box which can be used for general chat questions or for code fixes and refactors. To ask Cody for an inline code fix, select a code snippet, then type `/fix` plus your desired code change. Cody will generate edits, which you can accept or reject with the `Apply` button.

You can also use the or `/touch` command in the inline chat box if you'd like Cody to place its output in a new file.

![Example of Cody inline code fix ](https://storage.googleapis.com/sourcegraph-assets/website/Product%20Animations/GIFS/cody_inline_June23-sm.gif)

Examples of `/fix` instructions Cody can handle:

- Factor out any common helper functions (when multiple functions are selected)
- Use the imported CSS module's class `n`
- Extract the list item to a separate React component
- Handle errors in this code better
- Add helpful debug log statements
- Make this work (and yes, it often does work—give it a try!)

<div class="getting-started">
  <a class="demo text-center" target="_blank" href="https://twitter.com/sqs/status/1647673013343780864">View Demo</a>
</div>

## Commands

> **Commands** allow you to quickly run common actions. Commands are predefined, reusable prompts accessible by hotkey from within the [VS Code extension](https://marketplace.visualstudio.com/items?itemName=sourcegraph.cody-ai). Like autocomplete and chat, commands will search for context in your codebase to provide more contextually-aware and informed answers (or to generate more idiomatic code snippets).

Reviewer: ```suggestion
**Commands** allow you to run common actions quickly. Commands are predefined, reusable prompts accessible by hotkey from within the [VS Code extension](https://marketplace.visualstudio.com/items?itemName=sourcegraph.cody-ai). Like autocomplete and chat, commands will search for context in your codebase to provide more contextually aware and informed answers (or to generate more idiomatic code snippets).
```

The commands available in VS Code include:

- Document Code
- Explain Code
- Generate Unit Tests
- Code Smell

There are three ways to run a command in VS Code:

1. Type `/` in the chat bar. Cody will then suggest a list of available commands
2. Right click and select `"Cody"` > Choose a command from the list
3. Use the predefined command hotkey: `⌥` + `C` / `Alt` + `C`

> NOTE: This functionality is also available in the JetBrains extension under the name `Recipes`. To access it, navigate to the `Recipes` panel (next to the `Chat` panel), and you can find each available recipe as a button within the UI.

![Cody Commands in VS Code](https://storage.googleapis.com/sourcegraph-assets/Docs/cody-commands.png)

### Custom commands <span class="badge badge-experimental">Experimental</span>

> **Custom commands** let you to save your own quick actions and prompts for Cody based on your common workflows. They are defined in JSON format and let you call CLI tools, write custom prompts, and select context to be sent to Cody. This provides a flexible way to tailor Cody to your needs.

Reviewer: ```suggestion
**Custom commands** let you save your quick actions and prompts for Cody based on your common workflows. They are defined in JSON format and allow you to call CLI tools, write custom prompts, and select context to be sent to Cody. This provides a flexible way to tailor Cody to your needs.
```

> You can invoke custom comamnds with the same hotkey as predefined commands. Alternatively, you can right-click on selected code, open the Cody context menu and select the `Custom Commands (Experimental)` option.

Reviewer: ```suggestion
You can invoke custom commands with the same hotkey as predefined commands. Alternatively, you can right-click the selected code, open the Cody context menu, and select the `Custom Commands (Experimental)` option.
```

![Cody Custom Commands in VS Code](https://storage.googleapis.com/sourcegraph-assets/Docs/create-custom-commands.png)

### Defining commands in the `cody.json` file

> You can define custom commands for Cody in the `cody.json` file. To make commands only available for a specific project, create the `cody.json` file in the `.vscode` directory of that project. These workspace-specific custom commands will be available to you when you work on that project.

Reviewer: ```suggestion
You can define custom commands for Cody in the `cody.json` file. To make commands only available for a specific project, create the `cody.json` file in that project's `.vscode` directory. When you work on that project, these workspace-specific custom commands will be available.
```

> To make custom commands globally available across multiple projects, create a new `cody.json` file in the `.vscode` folder of your home directory. These global custom commands will be available for you in Cody in any workspace.

Reviewer: ```suggestion
To make custom commands globally available across multiple projects, create a new `cody.json` file in your home directory's `.vscode` folder. These global custom commands will be available in Cody in any workspace.
```

## Language support

> Cody uses LLMs trained on broad code, and we've found it to effectively support all common programming languages. However, the quality of autocompletions and other features may vary based on how well the underlying LLM model was trained on a given language.

Reviewer: ```suggestion
Cody uses LLMs trained on broad code, and we've found it to support all common programming languages effectively. However, the quality of autocompletion and other features may vary based on how well the underlying LLM model was trained in a given language.
```

> We've found Cody to work especially well with Python, Go, JavaScript, and TypeScript code.

Reviewer: ```suggestion

```

I think this is a repetitive line we can omit it.
